File: ABC/401-450/403/b.py
def main():
  t = input()
  u = input()
  for i in range(len(t)-len(u)+1):
    ok = True
    for j in range(len(u)):
      if t[i+j] != u[j] and t[i+j] != '?':
        ok = False
    if ok:
      print("Yes")
      exit(0)
  print("No")

if __name__ == '__main__':
  main()

File: ABC/401-450/403/b.cpp
#include <iostream>
using namespace std;

int main(){
  string t,u;
  cin >> t >> u;
  for(int i=0;i<t.size()-u.size()+1;i++){
    bool ok = true;
    for(int j=0;j<u.size();j++){
      if(t[i+j]=='?') continue;
      if(t[i+j] != u[j]) ok = false;
    }
    if(ok){
      cout << "Yes" << endl;
      return 0;
    }
  }
  cout << "No" << endl;
  return 0;
}

